### ISSUE TITLE
Training Movie Recommendation System

### ISSUE THREAD

**Issue by WillKoehrsen** [DOsinga/deep_learning_cookbook#20]
In notebook [4.2 Build a Recommender System](https://github.com/DOsinga/deep_learning_cookbook/blob/master/04.2%20Build%20a%20recommender%20system%20based%20on%20outgoing%20Wikipedia%20links.ipynb) the model is trained using `mse` as the loss function, treating the problem as regression with positive labels __1__ and negative labels __-1__. Would labeling the negative examples __0__ and training with a `binary_crossentropy` loss function (treating the problem as classification) be a valid approach? 

Also, do we have to merge the "link" and "movie" embeddings using a dot product (`Dot` layer)? Could we instead concatenate the embeddings, so the shape would be `[None, 100]` and then add an additional fully connected layer to make a prediction? Something like

```python

# Merge embeddings by concatenating
merged = Concatenate(name = 'merge', axis = 2)([link_embedding, movie_embedding])
merged = Reshape((1, ))(merged)

# Add fully connected layer for predictions
out = Dense(1, activation = None, name = 'output')(merged)
model = Model(inputs = [link, movie], outputs = out)

``` 

I understand that we're trying to create embeddings for movies so that "similar" movies have closer embeddings in terms of cosine distance. Embeddings make sense, but I guess my question is: are there different ways to train the network to produce the embeddings?

**Comment by DOsinga** [comment]
I guess we could try that too.

The intuition here is that you want to assign a vector to movie (the embedding) and we do this by using that embedding to predict something (the links). Movies with similar links would get a similar vector and so that's how the embedding should work. 

Concatenating would probably create a better model to predict the link, but it might not create as good an embedding.
